src/sentry/integrations/models/integration_feature.py

```diff
@@ -20,65 +20,68 @@
+        obj = int.__new__(cls, value)
+        obj._value_ = value
+        obj.feature_name = feature_name
+
+        return obj
 
     @classmethod
-    def as_choices(cls) -> tuple[tuple[int, str], ...]:
-        return (
-            (cls.API, "integrations-api"),
-            (cls.ISSUE_LINK, "integrations-issue-link"),
-            (cls.STACKTRACE_LINK, "integrations-stacktrace-link"),
-            (cls.EVENT_HOOKS, "integrations-event-hooks"),
-            (cls.PROJECT_MANAGEMENT, "integrations-project-management"),
-            (cls.INCIDENT_MANAGEMENT, "integrations-incident-management"),
-            (cls.FEATURE_FLAG, "integrations-feature-flag"),
-            (cls.ALERTS, "integrations-alert-rule"),
-            (cls.RELEASE_MANAGEMENT, "integrations-release-management"),
-            (cls.VISUALIZATION, "integrations-visualization"),
-            (cls.CHAT, "integrations-chat"),
-            (cls.SESSION_REPLAY, "integrations-session-replay"),
-        )
+    def from_str(cls, feature_name: str) -> Feature:
+        for value in cls:
+            if value.feature_name == feature_name:
+                return value
+        raise ValueError(f"Invalid feature name provided: {feature_name}")
 
     @classmethod
-    def as_str(cls, feature: int) -> str:
-        if feature == cls.ISSUE_LINK:
-            return "integrations-issue-link"
-        if feature == cls.STACKTRACE_LINK:
-            return "integrations-stacktrace-link"
-        if feature == cls.EVENT_HOOKS:
-            return "integrations-event-hooks"
-        if feature == cls.PROJECT_MANAGEMENT:
-            return "integrations-project-management"
-        if feature == cls.INCIDENT_MANAGEMENT:
-            return "integrations-incident-management"
-        if feature == cls.FEATURE_FLAG:
-            return "integrations-feature-flag"
-        if feature == cls.ALERTS:
-            return "integrations-alert-rule"
-        if feature == cls.RELEASE_MANAGEMENT:
-            return "integrations-release-management"
-        if feature == cls.VISUALIZATION:
-            return "integrations-visualization"
-        if feature == cls.CHAT:
-            return "integrations-chat"
-        if feature == cls.SESSION_REPLAY:
-            return "integrations-session-replay"
-        return "integrations-api"
+    def from_int(cls, value: int) -> Feature:
+        for enum_obj in cls:
+            if enum_obj.value == value:
+                return enum_obj
+        raise ValueError(f"Invalid feature value provided: {value}")
+
+    @classmethod
+    def as_str_choices(cls) -> list[str]:
+        return [obj.feature_name for obj in cls]
+
+    @classmethod
+    def as_int_choices(cls) -> list[int]:
+        return [obj.value for obj in cls]
+
+    @classmethod
+    def as_choices(cls) -> list[tuple[int, str]]:
+        return [(obj.value, obj.feature_name) for obj in cls]
+
+    def __str__(self) -> str:
+        return self.feature_name
```

Comment:
This feels like a bit of a kludge to provide the string implementation with the feature name. I suppose the alternative would be to create a new method called `to_feature_name_str`?

Reply:
yeah seeing how this is used, I feel like it would be better to be explicit with a getter